tools/run_tests/xds_k8s_test_driver/framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -582,20 +582,32 @@ def _delete_gamma_route(self, name, wait_for_deletion=True):
         logger.info("Deleting HTTPRoute %s", name)
         try:
             self.k8s_namespace.delete_gamma_route(name)
-        except (retryers.RetryError, k8s.NotFound) as e:
-            logger.info("HTTPRoute %s deletion failed: %s", name, e)
+        except k8s.NotFound:
```

Comment:
We should refactor most of these `_delete_*` methods by creating `_delete_resource`; similar to GCP APIs. But this is a fight for another day.